File: application/examples/readFileToTXT/readFile.cpp
#include<iostream>  
#include <windows.h>  
#include <string>  
#include<vector>
#include<sstream>
#include<fstream>
using namespace std;

vector<string> find(string FolderDirectory)
{
	vector<string> file;
	string szFileName = FolderDirectory + "/*.*";//指定目录下的直接的所有文件和目录

	WIN32_FIND_DATA FindData;
	HANDLE hFile;//句柄
	hFile = FindFirstFile(szFileName.c_str(), &FindData);

	if (hFile != INVALID_HANDLE_VALUE){
		if (FindData.dwFileAttributes != FILE_ATTRIBUTE_DIRECTORY){
			string FileDir = FolderDirectory + "\\";
			FileDir += FindData.cFileName;
			file.push_back(FileDir);
		}
		while (FindNextFile(hFile, &FindData)){
			if (FindData.dwFileAttributes != FILE_ATTRIBUTE_DIRECTORY){
				string FileDir = FolderDirectory + "\\";
				FileDir += FindData.cFileName;
				file.push_back(FileDir);
			}
		}
		FindClose(hFile);
	}
	else
	{
		printf("Invalid file directory !");
	}
	return file;
}
int main()
{
	string path = "E:\\OneDrive\\liurui-doc\\CCPR";

	vector<string> scan = find(path);
	ofstream output("C:\\Users\\liu-rui\\Desktop\\lib.txt", ios::in | ios::out | ios::trunc);
	for (auto &element : scan)
	{
		output << element.substr(path.size() + 1, 1000) << endl;
	}
	return 0;
}

File: application/examples/readFileToTXT/readFile.m
file=dir('E:\OneDrive\liurui-doc\CCPR');
fileSize=size(file);
fileout=fopen('matlabList.txt','w');
%     \n一般会操作系统被翻译成"行的结束"，即LF(Line-Feed)
%     \r会被翻译成"回车"，即CR(Cariage-Return)
%       对于文本文件的新行，在UNIX上，一般用\n(LF)来表示，Mac上用\r(CR)来表示，
%       Windows上是用\n\r(CR-LF)来表示。  
for i=1:fileSize(1);
    if(file(i).isdir==0)
        fprintf(fileout,'%s\r\n',file(i).name); 
    end
end